aiohttp/client_reqrep.py

```diff
@@ -82,6 +88,10 @@ def __init__(self, method, url, *,
         self.url = url.with_fragment(None)
         self.original_url = url
         self.method = method.upper()
+
+        if self.method in self.GET_METHODS and any((data, compress, chunked)):
+            raise ValueError('Method %s should not contain request-data.' % self.method)
```

Comment:
Yeap, that was a big surprise for me as well, but that's how things are. You can google the subject, iirc, on some w3c mailing list that case was discussed and considered as fine. Also, SHOULD is a strong recommendation, but it's not a mandatory behaviour as what MUST does.